fakeldap.py

```diff
@@ -440,22 +449,29 @@ def _add_s(self, dn, record):
             self.directory[dn] = entry
             return (105,[], len(self.calls), [])
 
-    def _simple_onelevel_search(self, base, filterstr):
-        search_attr_name, search_attr_value = filterstr[1:-1].split('=')
-
+    def _multiple_attrs_onelevel_search(self, base, attrs_conditions):
         result = []
         for dn, attrs in self.directory.items():
             if dn.endswith(',{}'.format(base)):
                 if ',' in dn.strip(',{}'.format(base)):
                     # This would mean going more than one level in.
                     continue
 
-                search_attr = attrs.get(search_attr_name)
-                if search_attr == search_attr_value:
+                match = True
+                for search_attr_name, search_attr_value in attrs_conditions:
+                    search_attr = attrs.get(search_attr_name)
+                    this_loop_match = False
+                    if search_attr == search_attr_value:
+                        this_loop_match = True
+                    elif isinstance(search_attr, list):  # if attr is in the format "attr_name": ["value", ]
+                        if len(search_attr) == 1 and search_attr[0] == search_attr_value:
+                            this_loop_match = True
+                    if not this_loop_match:
+                        match = False
+                        break
```

Comment:
I think we could write this a bit more simply by just doing like this?
```
                match = True
                for search_attr_name, search_attr_value in attrs_conditions:
                    search_attr = attrs.get(search_attr_name)
                    if search_attr == search_attr_value:
                        continue
                    elif isinstance(search_attr, list):  # if attr is in the format "attr_name": ["value", ]
                        if len(search_attr) == 1 and search_attr[0] == search_attr_value:
                            continue
                    else:
                        match = False
                        break
```

Reply:
This could be done with minor modification to your code:
```python
                match = True
                for search_attr_name, search_attr_value in attrs_conditions:
                    search_attr = attrs.get(search_attr_name)
                    if search_attr == search_attr_value:
                        continue
                    elif isinstance(search_attr, list):  # if attr is in the format "attr_name": ["value", ]
                        if len(search_attr) == 1 and search_attr[0] == search_attr_value:
                            continue
                        else:
                            match = False
                            break
                    else:
                        match = False
                        break

                if match:
                    result.append((dn, attrs))
```

not sure if that is better

Reply:
diff to current:

```diff
diff --git a/fakeldap.py b/fakeldap.py
index 03e8437..1d39aac 100644
--- a/fakeldap.py
+++ b/fakeldap.py
@@ -460,13 +460,15 @@ class MockLDAP(object):
                 match = True
                 for search_attr_name, search_attr_value in attrs_conditions:
                     search_attr = attrs.get(search_attr_name)
-                    this_loop_match = False
                     if search_attr == search_attr_value:
-                        this_loop_match = True
+                        continue
                     elif isinstance(search_attr, list):  # if attr is in the format "attr_name": ["value", ]
                         if len(search_attr) == 1 and search_attr[0] == search_attr_value:
-                            this_loop_match = True
-                    if not this_loop_match:
+                            continue
+                        else:
+                            match = False
+                            break
+                    else:
                         match = False
                         break
```

Reply:
also it would probably be better to iterate over `search_attr` :upside_down_face:

Reply:
ah, I got it and added option to match list at the same time:

308ecb10570d15d41c28d0bddfdfd0681e437dac

```python

                match = True
                for search_attr_name, search_attr_value in attrs_conditions:
                    search_attr = attrs.get(search_attr_name)
                    if search_attr == search_attr_value:
                        continue
                    elif isinstance(search_attr, list) and search_attr_value in search_attr:  # if attr is in the format "attr_name": ["value", ]
                        continue
                    else:
                        match = False
                        break

                if match:
                    result.append((dn, attrs))
```

Reply:
Ah, yup, that looks good, thanks for fixing the quick snippet i made!